### ISSUE TITLE
`reportInvalidTypeVarUse` false negative when default value can't be used to infer the generic

### ISSUE THREAD

**Issue by DetachHead** [DetachHead/basedpyright#596]
Code sample in [basedpyright playground](https://basedpyright.com/?typeCheckingMode=all&code=CYUwZgBGD20NoBUC6AKAbgQwDYFcQC4IsBLAZwBdEkIAfCYgO3IgF4IBGASggFoA%2BCAnwAoCGPFiAdNOHCMrKLBTcAxBBAAnDdA2ENIAA47yAVQYBrBtADuDAGoYNxDACMsIBAE8DIIA)

```python
def foo[T](value: list[T] | int = 1) -> T:
        ...

a = foo() # error: reportUnknownVariableType
```
